config/configmap.go

```diff
@@ -116,7 +116,8 @@ func (l *Map) Set(key string, value interface{}) {
 	// koanf doesn't offer a direct setting mechanism so merging is required.
 	merged := koanf.New(KeyDelimiter)
 	_ = merged.Load(confmap.Provider(map[string]interface{}{key: value}, KeyDelimiter), nil)
-	l.k.Merge(merged)
+	// Ignore error for now; it only fails with strict merge when the types are different
+	_ = l.k.Merge(merged)
```

Comment:
should Koanf decide to return an error for a different reason we would have no way to report the `Set` failed, so maybe we should return the error here instead?

Reply:
seems there's at least a couple of calls that could trigger an error in this func with `Load` just above this line, I would be in favour of returning an error from Set

Reply:
Opened #4467 for this, can you assign this to me @codeboten? I will do it on a separate PR to keep this one for non-breaking changes